### DISCUSSION TITLE
Is there a support of nested presets?

### DISCUSSION THREAD

**Discussion by abuklis** [renovatebot/renovate#19594] [Request Help] [answered]
Dear Renovate community,

My name is Anna and I am configuring Renovate currently for quite a big project in GitLab.

I want to have 3 layers of configured presets: common project Renovate preset (preset A), then team-specific config preset (preset B) extending preset A, and then configs per repository, extending preset B.

Is this even possible in Renovate?

I have done the following configuration:

1. created a directory `renovate-configs` in our common libraries GitLab repository
2. In this directory I have created 2 files: renovate-common-config.json and team-3-renovate-preset.json
<img width="285" alt="image" src="https://user-images.githubusercontent.com/20967414/209806046-647c9129-1bbf-4598-ac54-e43f8ee92e22.png">

 renovate-configs/renovate-common-config.json contents: 
```
 {"extends": [
    "config:base",
    ":rebaseStalePrs",
    ":updateNotScheduled"
  ]}
```
renovate-configs/team-3-renovate-preset.json contents:
```
{
  "extends": [
    "gitlab>path-to-common-libs-repo/libraries/renovate-configs:renovate-common-config"
  ],
  "schedule": ["before 9am every weekday"]
}
```
path-to-common-libs-repo hidden by me by privacy reasons.

3. Then, in completely another repository X in renovate.json I refer to team-3-renovate-preset.json. renovate.json contents from this repository X:
```
{
  "extends": [
    "gitlab>path-to-common-libs-repo/libraries/renovate-configs:team-3-renovate-preset.json"
  ]
}
```
Paths are correct. But with this setup I get the following error:
`Cannot find preset's package (gitlab>path-to-common-libs-repo/libraries/renovate-configs:team-3-renovate-preset.json)`

Could you please help me and tell what is wrong with this setup?
And again, is this is even possible to have preset inheritance?

Thank you in advance!

Best regards,
Anna.

**Comment by viceice** [comment, marked as answer]
replace the colon with double slash

**Reply by abuklis** [reply]
Hi, thank you, but it did not work. I replaced colon with a double slash in the files, and I am still getting the same error:
`  "validationError": "Cannot find preset's package (gitlab>path-to-common-libs-repo/libraries/renovate-configs//team-3-renovate-preset)"`

**Reply by viceice** [reply]
the double slash needs to be between repo and file path.

**Reply by abuklis** [reply]
Thanks a lot for your help. It worked out!
